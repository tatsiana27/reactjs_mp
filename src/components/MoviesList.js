import React, { Component } from 'react';
import { connect } from 'react-redux';
import { MovieItem } from './MovieItem';
import { EmptyResult } from './EmptyResult';

export class MoviesListUI extends Component {
    render() {
        const { data: movies = [] } = this.props.movies;
        return (
            <div className="movies-list">
                {movies.length > 0
                    ? movies.map(item =>
                        <MovieItem
                            key={item.id}
                            id={item.id}
                            title={item.title}
                            genre={item.genres.join(' & ')}
                            image={item.poster_path}
                            releaseDate={item.release_date}
                        />)
                    : <EmptyResult/>
                }

            </div>
        )
    };
}

export const mapStateToProps = ({movies}) => {
    return {
        movies: movies,
    };

};

export const mapDispatchToProps = (dispatch) => ({});

export const MoviesList = connect(mapStateToProps, mapDispatchToProps)(MoviesListUI);