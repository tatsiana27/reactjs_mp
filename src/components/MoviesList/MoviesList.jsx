import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { fetchMovies } from '../../modules/movies';
import { bindActionCreators } from 'redux';

import Movie from '../Movie';
import EmptyResult from '../EmptyResult';


class MoviesList extends Component {

  render () {
    const { movies } = this.props;
    return (
    <div className="movies-list">
      {movies.data
        ? movies.data.map(item => (
        <Movie item={item} key={item.id} />
      ))
        : <EmptyResult/>
      }
    </div>
    )
  }
};

MoviesList.propTypes = {
  movies: PropTypes.PropTypes.shape({}).isRequired,
  genre: PropTypes.string,
};

MoviesList.defaultProps = {
  movies: {}
};
const mapStateToProps = state => ({
  movies: state.movies.items,
  params: state.params,
});

const mapDispatchToProps = dispatch => bindActionCreators({
  fetchMovies,
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(MoviesList);
