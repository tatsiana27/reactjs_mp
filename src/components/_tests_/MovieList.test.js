import React from 'react';
import { MoviesList } from '../MoviesList';
import { shallow } from 'enzyme';

describe('MoviesList ', () => {
    test('should render MoviesList Component', () => {
        const wrapper = shallow(
            <MoviesList/>
        );

        expect(wrapper).toMatchSnapshot();
    });
});