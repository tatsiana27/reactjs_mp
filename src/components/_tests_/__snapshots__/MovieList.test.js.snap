// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesList  should render MoviesList Component 1`] = `
<div
  className="movies-list"
>
  <MovieItem
    genre="comedy"
    image="src/static/images/movie1.jpg"
    key="1"
    releaseDate="2010"
    title="Movie1"
  />
  <MovieItem
    genre="comedy"
    image="src/static/images/movie2.jpg"
    key="2"
    releaseDate="2011"
    title="Movie2"
  />
  <MovieItem
    genre="drama"
    image="src/static/images/movie1.jpg"
    key="3"
    releaseDate="2009"
    title="Movie3"
  />
</div>
`;
